solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -1316,44 +1298,30 @@ contract StandardPoolConverter is ConverterVersion, IConverter, ContractRegistry
     }
 
     /**
-     * @dev returns two samples required for calculating the network fee
+     * @dev returns the network wallet and fees
      *
-     * @param currProd  the current product of the balances
-     * @param prevProd  the previous product of the balances
+     * @param reserveBalance0 1st reserve balance
+     * @param reserveBalance1 2nd reserve balance
      *
-     * @return the square root of the current product of the balances
-     * @return the square root of the previous product of the balances
+     * @return the network wallet
+     * @return the network fee on the 1st reserve
+     * @return the network fee on the 2nd reserve
      */
-    function networkFeePoints(
-        uint256 currProd,
-        uint256 prevProd
-    ) private pure returns (uint256, uint256) {
-        return (
-            currProd > 0 ? MathEx.floorSqrt(currProd) : 0,
-            prevProd > 0 ? MathEx.floorSqrt(prevProd) : 0
-        );
-    }
+    function networkWalletAndFees(
+        uint256 reserveBalance0,
+        uint256 reserveBalance1
+    ) private view returns (address, uint256, uint256) {
+        uint256 prevPoint = floorSqrt(_reserveBalancesProduct);
+        uint256 currPoint = floorSqrt(reserveBalance0 * reserveBalance1);
+
+        if (currPoint > prevPoint) {
+            (address networkFeeWallet, uint32 networkFee) = INetworkSettings(addressOf(NETWORK_SETTINGS)).networkFeeParams();
+            uint256 n = (currPoint - prevPoint) * networkFee;
+            uint256 d = (currPoint * (PPM_RESOLUTION - networkFee)).add(prevPoint * networkFee * 2);
+            return (networkFeeWallet, reserveBalance0.mul(n).div(d), reserveBalance1.mul(n).div(d));
+        }
 
-    /**
-     * @dev returns the network fee ratio
-     *
-     * @param currPoint     the current product of the balances
-     * @param prevPoint     the previous product of the balances
-     * @param networkFee    the relative portion (in PPM units) that
-     * should be taken from the conversion fees accumulated in this pool
-     *
-     * @return numerator of the portion that should be taken from the balances of this pool
-     * @return denominator of the portion that should be taken from the balances of this pool
-     */
-    function networkFeeRatio(
-        uint256 currPoint,
-        uint256 prevPoint,
-        uint32 networkFee
-    ) private pure returns (uint256, uint256) {
-        return (
-            (currPoint - prevPoint) * networkFee,
-            (currPoint * (PPM_RESOLUTION - networkFee)).add(prevPoint * networkFee * 2)
-        );
+        return (address(0), 0, 0);
```

Comment:
I'd do an early return in this case as well, such that we always continue with "successful" flows, while exiting early on "unsuccessful" flows

Reply:
Done (FYI: increases gas cost, since LT and GT are cheaper than LTE and GTE).